manager/apis/app/v1alpha1/m4dmodule_types.go

```diff
@@ -87,6 +87,25 @@ type SupportedAction struct {
 	Level connectors.EnforcementAction_EnforcementActionLevel `json:"level,omitempty"`
 }
 
+// The spec of the endpoint exposed by the module
+type EndpointSpec struct {
+	// +required
+	Hostname string `json:"endpointName,omitempty"`
```

Comment:
style: match json tags to variable name (but start with lowercase).
That is: endpointName -> hostname

Reply:
Why is the host name part of the module specification? Modules are templates that are used in order to create helm installs of copies/services. Thus this is not a hostname of a service but an added string to a service name (that you're building as fqdn below)

Reply:
Since I'm intending to reuse this struct also for holding the "real" hostname (in the status), I think it doesn't matter.. If you have a better name I'm open to suggestions :)